packages/cli/src/util/telemetry/commands/env/pull.ts

```diff
@@ -0,0 +1,36 @@
+import { TelemetryClient } from '../..';
+
+export class EnvPullTelemetryClient extends TelemetryClient {
+  trackCliArgumentFilename(filename?: string) {
+    if (filename) {
+      this.trackCliArgument({
+        arg: 'filename',
+        value: this.redactedValue,
+      });
+    }
+  }
+
+  trackCliOptionEnvironment(environment?: string) {
+    if (environment) {
+      this.trackCliOption({
+        option: 'environment',
+        value: this.redactedValue,
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** track system environments

If this value is a system environment (`production`, `preview`), we should track that value. If it's any other value, we should use `this.redactedValue`.‏